utils/ensureUserForAttribution.ts

```diff
@@ -0,0 +1,34 @@
+import { Attribution, AttributionWithUser, isAttributionWithUser } from 'utils/types';
+
+import { getPartsOfFullName } from './names';
+
+/**
+ * Attaches a `user` prop to an attribution object if one doesn't already exist, i.e. if the
+ * attributed author isn't present on PubPub.
+ * TODO(ian): I think we should move towards doing this normalization in the database.
+ */
+export default (attribution: Attribution): AttributionWithUser => {
```

Comment:
This has been moved from `ensureUserForAttribution.js`. If someone can double-check that the logic matches I'd be very appreciative...when this breaks the results are usually catastrophic.

Reply:
Logic checks out!